src/WebJobs.Script.WebHost/Management/LinuxInstanceManager.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.File;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.WebHost.Configuration;
+using Microsoft.Azure.WebJobs.Script.WebHost.Management.LinuxSpecialization;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Management
+{
+    public abstract class LinuxInstanceManager : IInstanceManager
+    {
+        private readonly object _assignmentLock = new object();
+        private readonly ILogger _logger;
+        private readonly IMeshServiceClient _meshServiceClient;
+        private readonly IEnvironment _environment;
+        private readonly HttpClient _client;
+        private readonly IScriptWebHostEnvironment _webHostEnvironment;
+
+        private HostAssignmentContext _assignmentContext;
+
+        public LinuxInstanceManager(IHttpClientFactory httpClientFactory, IScriptWebHostEnvironment webHostEnvironment,
+            IEnvironment environment, ILogger<LinuxInstanceManager> logger, IMetricsLogger metricsLogger, IMeshServiceClient meshServiceClient)
+        {
+            _client = httpClientFactory?.CreateClient() ?? throw new ArgumentNullException(nameof(httpClientFactory));
+            _webHostEnvironment = webHostEnvironment ?? throw new ArgumentNullException(nameof(webHostEnvironment));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _meshServiceClient = meshServiceClient;
```

Comment:
Do we not want to handle if this is null too? Or we don't care?

Reply:
Yes, there are some instances when we inject a NullMeshServiceClient ([here](https://github.com/Azure/azure-functions-host/blob/ba112b76a93ffc145690fb9015eef41bcab71d88/src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs#L271)). So, we wouldn't want to throw for those cases